### ISSUE TITLE
no groundtruth fine match found for local regress

### ISSUE THREAD

**Issue by ShineFox** [zju3dv/EfficientLoFTR#63]
Hi author, I'm having the problem shown in the picture when I'm training directly with the training code on top of github.

I have trained loftr and other schemes on the same dataset without the above error, which seems to rule out the dataset. Therefore, I would like to ask you what is the reason for this.

![image](https://github.com/user-attachments/assets/9c92d7ca-adcd-4459-87e4-045e906ecaad)

**Comment by Sgbf1** [comment]
@ShineFox  Hello, may i ask u have u solved this problem? I also meet this error in training megadepth and  trained loftr and xoftr  without this
`No groundtruth fine match found for local regress: [['MegaDepth_v1\\0022\\dense0\\imgs\\427154679_de14c315f4_o.jpg'], ['MegaDepth_v1\\0022\\dense0\\imgs\\3553841868_b6ee93bf43_o.jpg']]
no fine matches to supervise`
and my loss is always NaN

**Comment by wyf2020** [comment]
Sorry for the late reply. It is normal to occasionally see this warning during training. This is because we have added additional constraints (consistent and covisible) to the ground truth construction in the two-stage training process. As a result, the fine match ground truth will be fewer compared to LOFTR. You can refer to the code [here](https://github.com/zju3dv/EfficientLoFTR/blob/089f6665722398007908977891f47f2c002f2aec/src/loftr/utils/supervision.py#L211C13-L211C32) (correct_0to1_f_mask) for more details. This warning will not affect the training process or cause NaN loss. For NaN loss, I recommend first trying to disable mixed precision training and using FP32 precision to avoid NaN loss.
